app/models/chunk.rb

```diff
@@ -2,7 +2,16 @@
 
 class Chunk < ApplicationRecord
   belongs_to :document
+  has_neighbors :embedding
 
   validates :chunk_index, presence: true
   validates :content, presence: true
+
+  before_create :generate_embedding
+
+  private
+
+  def generate_embedding
+    self.embedding = EmbeddingService.generate(content)
```

Comment:
I think we should probably do this in a background job since the embedding service could be slow here O(seconds). Will mean that we can decouple creating the embedding (and retry if needed) from the creation of the DB record.

Reply:
Yeah I am going to move the chuncking logic to the background once I figure out how will call the LLM client after chunking completion